From f55e73f3194b82bc7c7ece8b610c970c60627c5c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:44 -0400
Subject: [tools] lib/traceevent: Unregister handler when jbd2 plugin is is unloaded

Message-id: <1408953210-26343-277-git-send-email-jolsa@redhat.com>
Patchwork-id: 88927
O-Subject: [PATCH RHEL7.1 BZ1133083 276/322] tools lib traceevent: Unregister handler when jbd2 plugin is is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0247a967d4c1986821b3c7cbad324468fa6f2ae3
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:16 2014 +0900

description
===========
The function handlers should be unregistered when the plugin unloaded
otherwise they'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_jbd2.c b/tools/lib/traceevent/plugin_jbd2.c
index 2f93f81..0db714c 100644
--- a/tools/lib/traceevent/plugin_jbd2.c
+++ b/tools/lib/traceevent/plugin_jbd2.c
@@ -66,3 +66,12 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
            PEVENT_FUNC_ARG_VOID);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_print_function(pevent, process_jbd2_dev_to_name,
+      "jbd2_dev_to_name");
+
+ pevent_unregister_print_function(pevent, process_jiffies_to_msecs,
+      "jiffies_to_msecs");
+}
-- 
1.7.1